pkg/network/iface/link.go

```diff
@@ -14,23 +14,35 @@ import (
 	"k8s.io/utils/net/ebtables"
 )
 
-const defaultPVID = uint16(1)
+const (
+	defaultPVID = uint16(1)
+
+	EmptyIndex = 0
+	EmptyName  = ""
+
+	TypeLoopback = "loopback"
+	TypeDevice   = "device"
+	TypeBond     = "bond"
+)
 
 type Link struct {
 	link   netlink.Link
 	addr   []netlink.Addr
 	routes []netlink.Route
 }
 
-func GetLink(name string) (*Link, error) {
-	if name == "" {
-		return nil, fmt.Errorf("link name could not be empty string")
+func GetLink(name string, index int) (*Link, error) {
+	var l netlink.Link
+	var err error
+	if name != "" {
+		l, err = netlink.LinkByName(name)
+	} else {
+		l, err = netlink.LinkByIndex(index)
 	}
```

Comment:
Can we have `GetLinkByName` and `GetLinkByIndex` functions instead of always pass an empty param to this?